### PR TITLE
`prefer-spread`: Insert semicolon if needed

### LINKED ISSUE #254: `prefer-spread` fixer bug
`prefer-spread` fixer has a bug on no-semi code style

error code like this

```js
const foo = []
Array.from(arr).forEach(doSomething)
```

will fix to code like

```js
const foo = []
[...arr].forEach(doSomething)
```

no semi added before `[` on second line, breaks code

### PR DESCRIPTION
**Pull request by fisker**
I'm sure there is still other case not handled, but should good enough for #254

Fix #254

### CODE REVIEW

File: rules/prefer-spread.js [hunk 1]
```diff
@@ -14,7 +14,32 @@ const selector = [
 ].join('');
 
 const create = context => {
-	const getSource = node => context.getSourceCode().getText(node);
+	const sourceCode = context.getSourceCode();
+	const getSource = node => sourceCode.getText(node);
+
+	const needsSemicolon = node => {
```
Comment (sindresorhus) [inline rules/prefer-spread.js]: Should this be in `util`?

Comment (fisker) [reply rules/prefer-spread.js]: I tried to make a better util function for this, but seems need handle differently in different case.I don't think current implementation can use in other rule, at least not yet.

Comment (sindresorhus) [reply rules/prefer-spread.js]: Ok. Just something to keep in mind in case we need it in multiple places.

**Comment by fisker** [comment]
Screenshot_2020-03-05-22-20-46-024_com chrome canary

Not important, but  why the last  commit shows by you ? The `ava` one  by me ?  Github  bug?

**Comment by sindresorhus** [comment]
Yes, GitHub bug. I experienced it on another repo earlier too...

**Comment by sindresorhus** [comment]
Can you report it? I’m AFK.

**Comment by sindresorhus** [comment]
https://twitter.com/sindresorhus/status/1235621354528845830

**Pull request merged**